src/modules/ethereum/chains/ChainConfigStore.ts

```diff
@@ -1,18 +1,36 @@
 import { produce } from 'immer';
 import { equal } from 'src/modules/fast-deep-equal';
 import type { Chain } from 'src/modules/networks/Chain';
-import type { NetworkConfig } from 'src/modules/networks/NetworkConfig';
 import type { NetworkConfigMetaData } from 'src/modules/networks/Networks';
 import { PersistentStore } from 'src/modules/persistent-store';
 import { invariant } from 'src/shared/invariant';
 import { upsert } from 'src/shared/upsert';
+import type { NetworkConfig } from 'src/modules/networks/NetworkConfig';
+import { toAddEthereumChainParamer } from 'src/modules/networks/helpers';
+import { getNetworkByChainId } from 'src/modules/networks/networks-api';
+import type { Wallet } from 'src/background/Wallet/Wallet';
+import { INTERNAL_ORIGIN } from 'src/background/constants';
+import type { AddEthereumChainParameter } from '../types/AddEthereumChainParameter';
+import { getCustomNetworkId, isCustomNetworkId } from './helpers';
+
+export function maybeLocalChainId(id: string) {
+  return id.length === 21; // nanoid() standard length
+}
 
 export interface EthereumChainConfig extends NetworkConfigMetaData {
+  value: AddEthereumChainParameter;
+  id: string;
+}
+
+interface EtherenumNetworkConfig extends NetworkConfigMetaData {
   value: NetworkConfig;
 }
 
 export interface ChainConfig {
-  ethereumChains: EthereumChainConfig[];
+  ethereumChainConfigs?: EthereumChainConfig[];
+  migratedToV1_4?: boolean;
```

Comment:
What does this name mean? Why do we need such a complex flag?

Reply:
This is a dev name, I will change it. Just changed it for trying migration multiple times